consumer/src/main/java/com/flipkart/varadhi/consumer/MessageSrc.java

```diff
@@ -0,0 +1,15 @@
+package com.flipkart.varadhi.consumer;
+
+import java.util.concurrent.CompletableFuture;
+
+public interface MessageSrc {
+
+    /**
+     * Fills the messages array with the next messages to be processed. The array size dictates the max number of
+     * messages to return.
+     *
+     * @param messages
+     * @return count of messages filled in the array.
+     */
+    CompletableFuture<Integer> nextMessages(MessageTracker[] messages);
```

Comment:
We pass array (rather than List) so that we pre-allocate the memory upfront and re-use it?

Reply:
yes. that is the intention.